docs/knn/knn-score-script.md

```diff
@@ -0,0 +1,207 @@
+---
+layout: default
+title: Exact k-NN with Scoring Script
+nav_order: 2
+parent: KNN
+has_children: false
+has_math: true
+---
+
+# Exact k-NN with Scoring Script
+The k-NN plugin implements the Elasticsearch score script plugin that can be used to find the exact k-nearest neighbors to a given query point. Using the k-NN score script, a filter can be applied on an index before executing the nearest neighbor search. This is useful for dynamic search cases where the index body may vary based on other conditions. Because this approach executes a brute force search, it will not scale as well as the [Approximate approach](../approximate-knn). In some cases, it may be better to think about refactoring your workflow or index structure to use the Approximate approach instead of this approach.
+
+## Getting started with the score script
+
+Similar to approximate nearest neighbor search, in order to use the score script on a body of vectors, you first must create an index with one or more `knn_vector` fields. If you intend to just use the script score approach (and not the approximate approach) `index.knn` can be set to `false` and `index.knn.space_type` does not need to be set. The space type can be chosen during search. See the [spaces section](#spaces) to see what spaces the knn score script suppports. Here is an example that creates an index with two `knn_vector` fields:
```

Comment:
>> If you intend to just use the script score approach (and not the approximate approach) `index.knn` can be set to `false`

This should be put in **Bold** preferably kind of **Blurb** or may be just separate it as a new para and mention as Note. This is a very crucial point to pay attention.

Reply:
I also have as a tip here: https://github.com/opendistro/for-elasticsearch-docs/pull/396/files#diff-ba7ea95f84ff621a32b3ac90b917dc1956c4f8837b76f7106b4e9fb238fd1732R37